include/std/experimental/conqueue

```diff
@@ -440,6 +471,18 @@ struct buffer_queue<T, Alloc>::pop_sender {
             "async_pop: queue is empty, putting %p in the waiters queue\n",
             &op);
         self.waiters.push_back(&op);
+        // We are in the queue, set up the stop callback.
+        if constexpr (stoppable) {
+          op.stop_callback.emplace(stop_token, cancel_callback{op});
```

Comment:
You should try to avoid registering the stop-callback and the subsequent atomic ops if the stop-token returns `false` from `stop_possible()`.